schulcloud/SettingsViewController.swift

```diff
@@ -32,5 +42,58 @@ class SettingsViewController: UITableViewController {
             present(loginViewController, animated: true, completion: nil)
         }
     }
-
+    
+    func synchronizeToCalendar(switchView: UISwitch) {
+        
+        let newValue = switchView.isOn
+        if newValue {
+            CalendarEventHelper.requestCalendarPermission()
+            .flatMap { _ -> Future<[CalendarEvent], SCError> in
+                   return CalendarEventHelper.fetchCalendarEvent(inContext: managedObjectContext)
+            }
+            .flatMap { events -> Future< Void, SCError> in
+                    
+                guard let calendar = CalendarEventHelper.fetchCalendar() ?? CalendarEventHelper.createCalendar() else {
+                    return Future(error: .other("Can't access calendar") )
+                }
+                
+                do {
+                    try CalendarEventHelper.push(events: events, to: calendar)
+                } catch let error {
+                    return Future(error: .other(error.localizedDescription) )
+                }
+                return Future(value: Void() )
+            }
+            .onSuccess { _ in
+                DispatchQueue.main.async {
+                    currentEventKitSettings.isSynchonized = true
+                    switchView.isOn = true
+                }
+            }
+            .onFailure { error in
+                // TODO: Show error message as to why it failed
+                DispatchQueue.main.async {
+                    self.showErrorAlert(message: error.localizedDescription)
+                    currentEventKitSettings.isSynchonized = false
+                    switchView.isOn = false
+                }
+            }
+        } else {
+            do {
+                try CalendarEventHelper.deleteSchulcloudCalendar()
+                currentEventKitSettings.isSynchonized = false
+                switchView.isOn = false
+            } catch let error {
+                //TODO: Show error on why we could not delete the calendar
+                self.showErrorAlert(message: error.localizedDescription)
+                currentEventKitSettings.isSynchonized = true
+                switchView.isOn = true
+            }
+        }
+    }
+    
+    private func showErrorAlert(message: String) {
+        let alert = UIAlertController(title: "Somthing went wrong", message: message, preferredStyle: .alert)
+        self.present(alert, animated: true)
```

Comment:
The alert can't be dismissed. There is no action.

Reply:
Indeed, fixing that.